vncviewer/vncviewer.cxx

```diff
@@ -322,20 +322,30 @@ static const char* getlocaledir()
   return CMAKE_INSTALL_FULL_LOCALEDIR;
 #endif
 }
+
 static void init_fltk()
 {
   // Basic text size (10pt @ 96 dpi => 13px)
+  // FIXME: This is small by modern standards, but:
+  //        https://github.com/fltk/fltk/issues/371
   FL_NORMAL_SIZE = 13;
 
   // Select a FLTK scheme and background color that looks somewhat
   // close to modern systems
   Fl::scheme("gtk+");
-  Fl::background(220, 220, 220);
 
-  // macOS has a slightly brighter default background though
-#ifdef __APPLE__
+  // FIXME: Should get these from the system,
+  //        Fl::get_system_colors() is unfortunately not very capable
+  // FIXME: Should also handle dark mode
+  // FIXME: fl_contrast() also sucks since its calculations are utterly
+  //        wrong as they fail to account for sRGB curves. And even with
+  //        that fixed we get odd results with dark backgrounds, so
+  //        APCA should probably be used.
+  //        https://github.com/fltk/fltk/issues/370
   Fl::background(240, 240, 240);
-#endif
+  // GNOME uses (53,132,228), macOS (0, 122, 255), this is the lightest
+  // version of GNOME's that fl_contrast() won't screw up
+  Fl::set_color(FL_SELECTION_COLOR, 29, 113, 215);
```

Comment:
I didn't see an easy way to get the borders, and I think it will look odd without those.

Icons won't work as FLTK has no handling for inverting them when things get selected.

I also would have preferred aligning the labels on the left, but it looks horrible as FLTK has basically no margins.

All of this might be possible by doing a custom widget, but that is more work.

Reply:
![tigervnc_white_menu](https://user-images.githubusercontent.com/4117162/151556575-49dc3cad-381f-4569-b45f-3f0b07429ce0.png)

This is what it would look like with the white background, I had to add a white area left of the buttons to get the bottom left bit, but I think it looks fine without any borders. What do you think? Is it a problem that the Cancel/OK buttons look attached to the client area?

Reply:
![tigervnc_menu_with_border](https://user-images.githubusercontent.com/4117162/151563731-34fd3ae4-eeed-4ac8-888c-c482fc1f221c.png)

Added a border using:

    void Notebook::draw()
    {
      this->draw_box(FL_FLAT_BOX, labels->w(), 0, 1, h(), FL_DARK3);
      this->draw_children();
      Fl_Group::draw();
    }

Reply:
That looks odd. Looks like the buttons are part of the page, and are not global for everything.

But fair enough, if you want it that badly. ;)

Here's what a more complete implementation would look like:

![Screenshot from 2022-01-28 16-19-15](https://user-images.githubusercontent.com/6640916/151572986-e09a512d-9bd7-4838-913e-488910f0fbb9.png)

Reply:
Nice, I love that!